### PR TITLE
#185629329 : Switching to faraday

### PR DESCRIPTION
**Pull request by xinghengwang**
To support
- automatic retries. 
- uses net-http-persistent to keep alive and share pool of connections. 

This is a breaking change for these reasons: 
1. Faraday already decompresses gzip by default. 
2. For getting rules and getting app config, it make sense for the API layer to parse it into Hash (from json string) instead of let the Moesif_rack do it. Moesif_rack should just get things that are already parsed. Parsing should be done at API wrapper layer. 

Thus this will be a major version bump from 1.x.x to 2.0.1

### CODE REVIEW

**Comment by xinghengwang** [comment]
@keyur9  please review this

**Review by dgilling — APPROVED** [review]

**Comment by praves77** [comment]
@keyur9 ^^^ bump

File: lib/moesif_api/controllers/base_controller.rb [hunk 1]
```diff
@@ -1,20 +1,32 @@
...
+      }
+    end
+
+    def validate_response(context)
+      return if context.response.status_code.between?(200, 208) # [200,208] = HTTP OK
+
+      raise APIException.new 'HTTP Response Not OK', context
+    end
+
+    def decompress_and_parse(response)
```
Comment (keyur9) [inline lib/moesif_api/controllers/base_controller.rb]: may be we should remove this empty function

**Review by keyur9 — APPROVED** [review]
lgtm

**Review by praves77 — APPROVED** [review]

**Pull request merged**